config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml

```diff
@@ -76,6 +76,34 @@ presubmits:
           secretName: k8s-cip-test-prod-service-account
     annotations:
       testgrid-dashboards: sig-release-misc
+  - name: pull-cip-auditor-e2e
+    decorate: true
+    path_alias: "sigs.k8s.io/k8s-container-image-promoter"
+    skip_report: false
+    always_run: true
+    # Because we run e2e tests, we cannot run more than 1 instance at a time
+    # (the e2e test resources are not isolated across test runs).
+    max_concurrency: 1
+    branches:
+    - ^master$
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20200114-6d2c483-master
+        command:
+        - "./test-e2e/cip-auditor/entrypoint-from-container.sh"
+        env:
+        - name: CIP_E2E_KEY_FILE
+          value: "/etc/k8s-gcr-audit-test-prod-service-account/service-account.json"
+        volumeMounts:
+        - name: k8s-gcr-audit-test-prod-service-account-creds
+          mountPath: /etc/k8s-gcr-audit-test-prod-service-account
+          readOnly: true
+      volumes:
+      - name: k8s-gcr-audit-test-prod-service-account-creds
+        secret:
+          secretName: k8s-gcr-audit-test-prod-service-account
```

Comment:
Can we use workload identity, rather than secrets that can be stolen?

https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

Reply:
I'm sure we can, but (1) it doesn't really matter a ton if the secret is stolen because it only has powers to mess around in the test project (filled with meaningless test-only data) and (2) can we still not block on this?

Reply:
1) Merge https://github.com/kubernetes/test-infra/pull/15984
2) Delete the volumes, volume mounts, env 
3) Add `serviceAccountName: k8s-gcr-audit-test-prod` to the pod spec (aka between L89 and L90)

Everything at this point is declarative and self-managed. AKA the owner of the service account will have to run 

```
gcloud iam service-accounts add-iam-policy-binding \
  --role roles/iam.workloadIdentityUser \
  --member "serviceAccount:[PROJECT_ID].svc.id.goog[[K8S_NAMESPACE]/[KSA_NAME]]" \
  [GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com
```

From the above docs, and #15984 will have to merge, but once that happens `gcloud auth list` inside the pod will show that things authenticate to GCP as `[GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com`

Alternatively, yes, we can definitely do the service account thing. Can you hand that over to @Katharine tomorrow if you want to go that route?

Reply:
#15984 looks very promising and since we got this far, I would like to try it out (the config example looks almost too easy/good to be true...!). I was going to ask how do I figure out the PROJECT_ID, K8S_NAMESPACE/KSA_NAME but it looks like you already answered those questions by adding the initial config for the GSA I wanted to add in this PR (https://github.com/kubernetes/test-infra/pull/15984/files#diff-2ee02ef3182b8522bfa83a6c7a2d6978R6-R8) --- bravo!

Anyway I'll edit this PR as you instructed. Brb